av1an-core/src/lib.rs

```diff
@@ -180,6 +180,31 @@ impl Input {
     })
   }
 
+  pub fn transfer_function_params_adjusted(
+    &self,
+    enc_params: &[String],
+  ) -> anyhow::Result<TransferFunction> {
+    if enc_params.iter().any(|p| {
+      let p = p.to_lowercase();
+      p == "pq" || p.ends_with("=pq") || p.ends_with("smpte2084")
+    }) {
+      return Ok(TransferFunction::SMPTE2084);
+    }
+    if enc_params.iter().any(|p| {
+      let p = p.to_lowercase();
```

Comment:
My only (extremely minor) nitpick is that these 2 `to_lowercase` function calls could instead be `to_ascii_lowercase`, which are theoretically a lot faster because they don't have to load values from a huge utf8 lookup table.

Reply:
Technically true. It won't affect overall performance as this function is called at most twice per encode, but since it's still in PR might as well make it the better version.